Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -831,3 +834,41 @@ void MainWindow::OnBootGameCubeIPL(DiscIO::Region region)
 {
   StartGame(std::make_unique<BootParameters>(BootParameters::IPL{region}));
 }
+
+void MainWindow::OnImportNANDBackup()
+{
+  auto response = QMessageBox::question(
+      this, tr("Question"),
+      tr("Merging a new NAND over your currently selected NAND will overwrite any channels "
+         "and savegames that already exist. This process is not reversible, so it is "
+         "recommended that you keep backups of both NANDs. Are you sure you want to "
+         "continue?"));
+
+  if (response == QMessageBox::No)
+    return;
+
+  QString file = QFileDialog::getOpenFileName(this, tr("Select the save file"), QDir::currentPath(),
+                                              tr("BootMii NAND backup file (*.bin);;"
+                                                 "All Files (*)"));
+
+  if (file.isEmpty())
+    return;
+
+  QProgressDialog dialog(this);
+  dialog.setMinimum(0);
+  dialog.setMaximum(0);
+  dialog.setLabelText(tr("Importing NAND backup"));
+  dialog.setCancelButton(nullptr);
+  dialog.show();
+
+  auto beginning = QDateTime::currentDateTime().toSecsSinceEpoch();
+
+  DiscIO::NANDImporter().ImportNANDBin(file.toStdString(), [&dialog, beginning] {
```

Comment:
This should run in a separate thread so the UI stays responsive. DolphinQt2/WiiUpdate.cpp has an example of how to do that.